File: orchestrator-api/src/main/java/com/example/gwp/orchestrator/adapter/kubernetes/MockJobDispatcher.java
package com.example.gwp.orchestrator.adapter.kubernetes;

import com.example.gwp.orchestrator.application.ImageLogMask;
import com.example.gwp.orchestrator.domain.Job;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class MockJobDispatcher implements JobDispatcher {

    @Override
    public String dispatch(Job job) {
        String name = "mock-job-" + job.getId();
        log.info("[mock] dispatch jobId={} image={} gpu={} → {}",
                job.getId(), ImageLogMask.mask(job.getImage()), job.getGpuCount(), name);
        return name;
    }

    @Override
    public void cancel(String k8sJobName) {
        log.info("[mock] cancel name={}", k8sJobName);
    }
}

File: orchestrator-api/src/main/kotlin/com/example/gwp/orchestrator/adapter/kubernetes/MockJobDispatcher.kt
package com.example.gwp.orchestrator.adapter.kubernetes

import com.example.gwp.orchestrator.application.ImageLogMask
import com.example.gwp.orchestrator.domain.Job
import org.slf4j.LoggerFactory

class MockJobDispatcher : JobDispatcher {

    override fun dispatch(job: Job): String {
        val name = "mock-job-" + job.id
        log.info(
            "[mock] dispatch jobId={} image={} gpu={} → {}",
            job.id, ImageLogMask.mask(job.image), job.gpuCount, name,
        )
        return name
    }

    override fun cancel(k8sJobName: String) {
        log.info("[mock] cancel name={}", k8sJobName)
    }

    companion object {
        private val log = LoggerFactory.getLogger(MockJobDispatcher::class.java)
    }
}
